pkg/cached/cache_test.go

```diff
@@ -31,139 +31,138 @@ import (
-			return cached.NewResultErr[[]byte](errors.New("source error"))
-		} else {
-			return cached.NewResultOK([]byte("source"), "source")
+			return nil, "", errors.New("source error")
 		}
+		return []byte("source"), "source", nil
 	})
-	if err := source.Get().Err; err != nil {
+	if _, _, err := source.Get(); err != nil {
 		t.Fatalf("unexpected error: %v", err)
 	}
-	if err := source.Get().Err; err == nil {
+	if _, _, err := source.Get(); err == nil {
 		t.Fatalf("expected error, found none")
 	}
-	if err := source.Get().Err; err != nil {
+	if _, _, err := source.Get(); err != nil {
 		t.Fatalf("unexpected error: %v", err)
 	}
-	if err := source.Get().Err; err == nil {
+	if _, _, err := source.Get(); err == nil {
 		t.Fatalf("expected error, found none")
 	}
 	if count != 4 {
 		t.Fatalf("Expected function called 4x, called: %v", count)
 	}
 }
 
-func TestStaticSource(t *testing.T) {
+func TestOnce(t *testing.T) {
 	count := 0
-	source := cached.NewStaticSource(func() cached.Result[[]byte] {
+	source := cached.Once(cached.Func(func() ([]byte, string, error) {
 		count += 1
-		return cached.NewResultOK([]byte("source"), "source")
-	})
-	if err := source.Get().Err; err != nil {
+		return []byte("source"), "source", nil
+	}))
+	if _, _, err := source.Get(); err != nil {
 		t.Fatalf("unexpected error: %v", err)
 	}
-	if err := source.Get().Err; err != nil {
+	if _, _, err := source.Get(); err != nil {
 		t.Fatalf("unexpected error: %v", err)
 	}
 	if count != 1 {
 		t.Fatalf("Expected function called once, called: %v", count)
 	}
 }
 
-func TestStaticSourceError(t *testing.T) {
+func TestOnceError(t *testing.T) {
 	count := 0
-	source := cached.NewStaticSource(func() cached.Result[[]byte] {
+	source := cached.Once(cached.Func(func() ([]byte, string, error) {
 		count += 1
-		return cached.NewResultErr[[]byte](errors.New("source error"))
-	})
-	if err := source.Get().Err; err == nil {
+		return nil, "", errors.New("source error")
+	}))
+	if _, _, err := source.Get(); err == nil {
 		t.Fatalf("expected error, found none")
 	}
-	if err := source.Get().Err; err == nil {
+	if _, _, err := source.Get(); err == nil {
 		t.Fatalf("expected error, found none")
 	}
 	if count != 1 {
 		t.Fatalf("Expected function called once, called: %v", count)
 	}
 }
 
-func TestResultData(t *testing.T) {
-	source := cached.NewResultOK([]byte("source"), "source")
-	if err := source.Get().Err; err != nil {
+func TestNewResult(t *testing.T) {
+	source := cached.Static([]byte("source"), "source")
+	if _, _, err := source.Get(); err != nil {
 		t.Fatalf("unexpected error: %v", err)
 	}
-	if err := source.Get().Err; err != nil {
+	if _, _, err := source.Get(); err != nil {
 		t.Fatalf("unexpected error: %v", err)
 	}
 }
 
-func TestResultError(t *testing.T) {
-	source := cached.NewResultErr[[]byte](errors.New("source error"))
-	if err := source.Get().Err; err == nil {
+func TestNewError(t *testing.T) {
+	source := cached.Result[[]byte]{Err: errors.New("source error")}
```

Comment:
Is this fine as a way for instantiating errors? I noticed that the `NewError` and `Errorf` functions were removed.

Reply:
you usually won't have to construct these results as we have switches to tripples as return values.

Have fixed these two tests to reference what they do: test `Get` of `Result`.

Done.